GOVERNANCE.md

```diff
@@ -61,19 +61,67 @@ maintainer has already discussed the candidacy with the candidate and a
+## Adding subprojects
+
+containerd subprojects are divided into two flavors currently: **core** and
+**non-core**. Most of the repositories within the containerd GitHub organization are
+"core" to the delivery of the containerd runtime project's releases. For example,
+the Golang implementations of the cgroups, runc, and console APIs, the ttrpc
+implementation, and various plugins like the CRI implementation. These, among a
+handful of other repositories, are all **core** and required pieces which culminate
+in officially supported and maintained releases of containerd.
+
+Non-core projects have strong affiliation with the containerd runtime, but
+operate in some ways similar to the traditional `contrib/` directory in many
+open source projects. They are most likely separately maintained, and potentially
+have a different release cycle and support guarantee than the main containerd project
+and its core subprojects. They may be written in other languages and therefore
+require different skills, developer tools, and CI systems than the core projects.
+For these reasons, non-core subprojects will be treated differently than core
+projects within the governance of the containerd organization.
+
+Both core and non-core subprojects must adhere to the CNCF
+[charter](https://www.cncf.io/about/charter/) and mission.
+
+### Adding core subprojects
+
+New core subprojects can request to be added to the containerd GitHub
+organization by submitting a project proposal via public forum (a
+`containerd/project` GitHub issue is the easiest way to provide this proposal).
+The existing maintainers are given five business days to discuss the new
 project, raise objections and cast their vote. Projects must be approved by at
 least 66% of the current maintainers by adding their vote.
 
 If a project is approved, a maintainer will add the project to the containerd
 GitHub organization, and make an announcement on a public forum.
 
+To make this transition to being a core subproject easier, please prepare your
+repository with the DCO signoff checks, the fileheader check, and by adding the
+appropriate Apache 2 license file to your project root as well as the boilerplate
+header to all applicable source files.
+
+### Adding non-core subprojects
+
+Non-core subprojects will also submit a project proposal via public forum, and
+should state that the project is expected to be **non-core**.
+
+The proposal should include a proposed list of maintainers who will manage
+the non-core project and provide general information on support, releases,
+stability, and any additional detail useful for the containerd maintainers to
+understand the scope and nature of the project.
+
+The existing maintainers are given five business days to discuss the new
+project, raise objections and cast their vote. Projects must be approved by at
+least 50% of the current maintainers by adding their vote.
+
+If a project is approved, a maintainer will add the project to the containerd
```

Comment:
All projects will be in the same org, then how do we clearly show that a project is core or not? And do we need to clearly show that?

Reply:
Yeah--was wondering if there was a GH feature (tagging a repo with some visible tag/annotation?), but assuming there isn't, I think we can provide boilerplate (like what is at the bottom of all subprojects right now) that will be different for core and non-core projects. I will make that clear in the next revision of this text.

Reply:
Just something in the readme works fine.

Reply:
Something in the readme SG, just need to make sure we have a convention and all subprojects follow that.

Reply:
Added section to `README.md` similar to the existing boilerplate with specified wording changes/additions for non-core projects. The end of each "adding a subproject" section now points directly to that section to provide guidance for what to add to your `README.md`.